File: tests/rosetta/transpiler/Java/check-that-file-exists.java
public class Main {

    static void printStat(java.util.Map<String,Boolean> fs, String path) {
        if (((Boolean)(fs.containsKey(path)))) {
            if (((boolean)(fs).getOrDefault(path, false))) {
                System.out.println(path + " is a directory");
            } else {
                System.out.println(path + " is a file");
            }
        } else {
            System.out.println("stat " + path + ": no such file or directory");
        }
    }

    static void main() {
        java.util.Map<String,Boolean> fs = ((java.util.Map<String,Boolean>)(new java.util.LinkedHashMap<String, Boolean>()));
fs.put("docs", true);
        for (String p : new String[]{"input.txt", "/input.txt", "docs", "/docs"}) {
            printStat(fs, p);
        }
    }
    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            main();
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        rt.gc();
        return rt.totalMemory() - rt.freeMemory();
    }
}

File: tests/rosetta/transpiler/Clojure/check-that-file-exists.clj
(ns main (:refer-clojure :exclude [printStat main]))

(require 'clojure.set)

(defn in [x coll]
  (cond (string? coll) (clojure.string/includes? coll x) (map? coll) (contains? coll x) (sequential? coll) (some (fn [e] (= e x)) coll) :else false))

(defn padStart [s w p]
  (loop [out (str s)] (if (< (count out) w) (recur (str p out)) out)))

(def nowSeed (atom (let [s (System/getenv "MOCHI_NOW_SEED")] (if (and s (not (= s ""))) (Integer/parseInt s) 0))))

(declare printStat main)

(declare main_fs)

(defn printStat [printStat_fs printStat_path]
  (if (in printStat_path printStat_fs) (if (get printStat_fs printStat_path) (println (str printStat_path " is a directory")) (println (str printStat_path " is a file"))) (println (str (str "stat " printStat_path) ": no such file or directory"))))

(defn main []
  (do (def main_fs {}) (def main_fs (assoc main_fs "docs" true)) (doseq [p ["input.txt" "/input.txt" "docs" "/docs"]] (printStat main_fs p))))

(defn -main []
  (let [rt (Runtime/getRuntime)
    start-mem (- (.totalMemory rt) (.freeMemory rt))
    start (System/nanoTime)]
      (main)
      (System/gc)
      (let [end (System/nanoTime)
        end-mem (- (.totalMemory rt) (.freeMemory rt))
        duration-us (quot (- end start) 1000)
        memory-bytes (Math/abs ^long (- end-mem start-mem))]
        (println (str "{\n  \"duration_us\": " duration-us ",\n  \"memory_bytes\": " memory-bytes ",\n  \"name\": \"main\"\n}"))
      )
    ))

(-main)
